plugins/com.google.cloud.tools.eclipse.swtbot/src/com/google/cloud/tools/eclipse/swtbot/SwtBotProjectActions.java

```diff
@@ -0,0 +1,313 @@
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.tools.eclipse.swtbot;
+
+import static org.eclipse.swtbot.swt.finder.matchers.WidgetMatcherFactory.widgetOfType;
+
+import org.eclipse.swt.widgets.MenuItem;
+import org.eclipse.swt.widgets.Tree;
+import org.eclipse.swt.widgets.Widget;
+import org.eclipse.swtbot.eclipse.finder.SWTWorkbenchBot;
+import org.eclipse.swtbot.eclipse.finder.widgets.SWTBotView;
+import org.eclipse.swtbot.swt.finder.exceptions.WidgetNotFoundException;
+import org.eclipse.swtbot.swt.finder.finders.ContextMenuHelper;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotMenu;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotTable;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotTree;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotTreeItem;
+
+/**
+ * SWTBot utility methods that perform general workbench actions.
+ */
+public final class SwtBotProjectActions {
+
+  private static final String SOURCE_FOLDER = "src";
+
+  /**
+   * Creates a Java class with the specified name.
+   *
+   * @param bot The SWTWorkbenchBot
+   * @param projectName The name of the project the class should be created in
+   * @param packageName The name of the package the class should be created in
+   * @param className The name of the class to be created
+   */
+  public static void createJavaClass(final SWTWorkbenchBot bot, String projectName,
+      String packageName, final String className) {
+    SWTBotTreeItem project = SwtBotProjectActions.selectProject(bot, projectName);
+    selectProjectItem(project, SOURCE_FOLDER, packageName).select();
+    SwtBotTestingUtilities.performAndWaitForWindowChange(bot, new Runnable() {
+      @Override
+      public void run() {
+        MenuItem menuItem = ContextMenuHelper.contextMenu(getProjectRootTree(bot), "New", "Class");
+        new SWTBotMenu(menuItem).click();
+      }
+    });
+
+    SwtBotTestingUtilities.performAndWaitForWindowChange(bot, new Runnable() {
+      @Override
+      public void run() {
+        bot.activeShell();
+        bot.textWithLabel("Name:").setText(className);
+        SwtBotTestingUtilities.clickButtonAndWaitForWindowChange(bot, bot.button("Finish"));
+      }
+    });
+  }
+
+  /**
+   * Creates a Java project with the specified name.
+   *
+   * @param bot the SWTWorkbenchBot
+   * @param projectName the project name
+   */
+  public static void createJavaProject(SWTWorkbenchBot bot, String projectName) {
+    // Open Java Perspective
+    bot.perspectiveById("org.eclipse.jdt.ui.JavaPerspective").activate();
```

Comment:
There's already a method to open the Java perspective in `SwtBotMenuActions`. Can we consolidate that code with this and call that method instead?
Also comparing this code to the one opening the Debug perspective in `SwtBotMenuActions`, this is much simpler.